doc/modules/decomposition.rst

```diff
@@ -278,6 +257,88 @@ factorization, while larger values shrink many coefficients to zero.
      R. Jenatton, G. Obozinski, F. Bach, 2009
 
 
+.. _kernel_PCA:
+
+Kernel Principal Component Analysis (kPCA)
+==========================================
+
+Exact Kernel PCA
+----------------
+
+:class:`KernelPCA` is an extension of PCA which achieves non-linear
+dimensionality reduction through the use of kernels (see :ref:`metrics`). It
+has many applications including denoising, compression and structured
+prediction (kernel dependency estimation). :class:`KernelPCA` supports both
+``transform`` and ``inverse_transform``.
+
+.. figure:: ../auto_examples/decomposition/images/sphx_glr_plot_kernel_pca_001.png
+    :target: ../auto_examples/decomposition/plot_kernel_pca.html
+    :align: center
+    :scale: 75%
+
+.. topic:: Examples:
+
+    * :ref:`sphx_glr_auto_examples_decomposition_plot_kernel_pca.py`
+
+.. _Randomized_kPCA:
+
+Kernel PCA using randomized SVD
```

Comment:
Call this "Choice of solver"?

It'd be nice to also briefly discuss the other ones, but don't feel obliged

Reply:
Good proposition, I'll work on that tomorrow

Reply:
Actually in PCA it is named like this. https://scikit-learn.org/stable/modules/decomposition.html#pca-using-randomized-svd

But I agree that your proposal makes more sense, so I'll update the title still.

Reply:
Done. I discuss both randomized and dense, but not arpack as I do not know it well enough.